src/lib/supercomputer.ts

```diff
@@ -132,28 +92,28 @@ export class Supercomputer {
       ? childrenObject[childName].inputs
       : inputs;
 
+    const inputsCopy = specificInputs.map((input: any) => ({...input}));
+
     const childrenConfig = childrenObject[childName].config || {};
 
-    const enrichedInputs = this.enrichInputs(
-      specificInputs,
-      config,
-      childrenConfig
+    const outputs = await pipeline.reduce(
```

Comment:
Observatory class was replaced with this `reduce` function. 
I believe now it has less complexity and more readability.